modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/RowHandler.java

```diff
@@ -104,4 +99,50 @@ interface RowFactory<RowT> {
          */
         RowT create(InternalTuple tuple);
     }
+
+    /**
+     * A builder to create rows. It uses the schema provided by an instance of row factory that created it.
+     *
+     * <pre>
+     *     // Create a row builder.
+     *     var rowBuilder = rowFactory.rowBuilder();
+     *     ...
+     *     // Call build() after all fields have been set.
+     *     var row1 = rowBuilder.build();
+     *     // Call reset() to cleanup builder's state.
+     *     rowBuilder.reset();
+     * </pre>
+     */
+    interface RowBuilder<RowT> {
+
+        /**
+         * Adds a field to the current row.
+         *
+         * @param value Field value.
+         * @return this.
+         */
+        RowBuilder<RowT> addField(@Nullable Object value);
+
+        /** Creates a new row from a previously added fields. */
+        RowT build();
+
+        /**
+         * Resets the state of this builder.
+         */
+        void reset();
+
+        /**
+         * Creates a new row and resets the state of this builder. This is a shorthand for:
+         * <pre>
+         *     Row row = builder.build();
+         *     builder.reset();
+         *     return row;
+         * </pre>
+         */
+        default RowT buildAndReset() {
```

Comment:
Do we use in production code `build` and/or `reset`?
Can we remove `reset` and `buildAndReset` and keep single `build` method?

p.s. I checked such refactoring in tests and it seems to work well, maybe we can get rid of unnecessary methods?

Reply:
What I mean is the following:
remove `reset` from interface
remove `build` from interface
rename buildAndReset to `build`

Reply:
In some sense I agree. But I still don’t understand what’s the point of keeping methods in the interface that we don’t use (build/reset). Just to explain what buildAndReset does?

Reply:
In general I'm not against the current implementation,